File: play-java-streaming-example/app/controllers/HomeController.scala
package controllers

import javax.inject.Inject

import play.api.mvc._
import play.api.routing._

class HomeController @Inject()(cc: ControllerComponents) extends AbstractController(cc) {

  def index(): Action[AnyContent] = Action { implicit request =>
    Ok(views.html.index())
  }

  def javascriptRoutes: Action[AnyContent] = Action { implicit request =>
    Ok(
      JavaScriptReverseRouter("jsRoutes")(
        routes.javascript.JavaEventSourceController.streamClock
      )
    ).as("text/javascript")
  }
}

File: java/streaming/app/controllers/HomeController.java
package controllers;

import javax.inject.Inject;

import play.routing.*;

import play.mvc.Controller;
import play.mvc.Http;
import play.mvc.Result;

public class HomeController extends Controller {

  public Result index(final Http.Request request) {
    return ok(views.html.index.render(request));
  }

  public Result javascriptRoutes(final Http.Request request) {
    return ok(
      JavaScriptReverseRouter.create(
        "jsRoutes",
        "jQuery.ajax",
        request.host(),
        routes.javascript.JavaEventSourceController.streamClock()
      )
    ).as("text/javascript");
  }
}
